fem/gslib.cpp

```diff
@@ -837,6 +837,155 @@ void FindPointsGSLIB::InterpolateGeneral(const GridFunction &field_in,
+                                   const double bb_t, const double newt_tol,
+                                   const int npt_max)
+{
+   MFEM_VERIFY(m.GetNodes() != NULL, "Mesh nodes are required.");
+   MFEM_VERIFY(m.GetNumGeometries(m.Dimension()) == 1,
+               "Mixed meshes are not currently supported in FindPointsGSLIB.");
+
+   //FreeData if OversetFindPointsGSLIB::Setup has been called already
+   if (setupflag) { FreeData(); }
+
+   crystal_init(cr, gsl_comm);
+   mesh = &m;
+   dim  = mesh->Dimension();
+   const FiniteElement *fe = mesh->GetNodalFESpace()->GetFE(0);
+   unsigned dof1D = fe->GetOrder() + 1;
+   int gt      = fe->GetGeomType();
+
+   if (gt == Geometry::TRIANGLE || gt == Geometry::TETRAHEDRON ||
+       gt == Geometry::PRISM)
+   {
+      GetSimplexNodalCoordinates();
+   }
+   else if (gt == Geometry::SQUARE || gt == Geometry::CUBE)
+   {
+      GetQuadHexNodalCoordinates();
+   }
+   else
+   {
+      MFEM_ABORT("Element type not currently supported in FindPointsGSLIB.");
+   }
+
+   MFEM_ASSERT(meshid>=0, " The ID should be greater than or equal to 0.");
+
+   const int pts_cnt = gsl_mesh.Size()/dim,
+             NEtot = pts_cnt/(int)pow(dof1D, dim);
+
+   distfint.SetSize(pts_cnt);
+   if (!gfmax)
+   {
+      distfint = 0.;
+   }
+   else
+   {
+      GetNodeValues(*gfmax, distfint);
+   }
+   u_meshid = (unsigned int)meshid;
+
+   if (dim == 2)
+   {
+      unsigned nr[2] = { dof1D, dof1D };
+      unsigned mr[2] = { 2*dof1D, 2*dof1D };
+      double * const elx[2] = { &gsl_mesh(0), &gsl_mesh(pts_cnt) };
+      fdata2D = findptsms_setup_2(gsl_comm, elx, nr, NEtot, mr, bb_t,
+                                  pts_cnt, pts_cnt, npt_max, newt_tol,
+                                  &u_meshid, &distfint(0));
+   }
+   else
+   {
+      unsigned nr[3] = { dof1D, dof1D, dof1D };
+      unsigned mr[3] = { 2*dof1D, 2*dof1D, 2*dof1D };
+      double * const elx[3] =
+      { &gsl_mesh(0), &gsl_mesh(pts_cnt), &gsl_mesh(2*pts_cnt) };
+      fdata3D = findptsms_setup_3(gsl_comm, elx, nr, NEtot, mr, bb_t,
+                                  pts_cnt, pts_cnt, npt_max, newt_tol,
+                                  &u_meshid, &distfint(0));
+   }
+   setupflag = true;
+   overset   = true;
+}
+
+void OversetFindPointsGSLIB::FindPoints(const Vector &point_pos,
+                                        Array<unsigned int> &point_id)
+{
+   MFEM_VERIFY(setupflag, "Use OversetFindPointsGSLIB::Setup before "
+               "finding points.");
+   MFEM_VERIFY(overset, " Please setup FindPoints for overlapping grids.");
+   points_cnt = point_pos.Size() / dim;
+   unsigned int match = 0; // Don't find points in the mesh if point_id = mesh_id
+
+   gsl_code.SetSize(points_cnt);
+   gsl_proc.SetSize(points_cnt);
+   gsl_elem.SetSize(points_cnt);
+   gsl_ref.SetSize(points_cnt * dim);
+   gsl_dist.SetSize(points_cnt);
+
+   if (dim == 2)
+   {
+      const double *xv_base[2];
+      xv_base[0] = point_pos.GetData();
```

Comment:
So if this is just intended to be used on the CPU/Host for now, I would suggest making use of `HostRead/HostReadWrite` here and the other places instead of `GetData` just to make sure data is kept in-sync with what's potentially on the device.